### DISCUSSION TITLE
Too many new releases 👨🏻‍💻

### DISCUSSION THREAD

**Discussion by morellodev** [TanStack/query#1202] [General]
Hi @tannerlinsley,

you know I really appreciate your hard work and I am so grateful for your amazing libraries 🙏🏻 I am opening this discussion because you are such a workaholic (*just joking*)! You are releasing 3 main releases in less than a year, each one with a set of breaking changes (v0.x ➡️ v1.x ➡️ v2.x ➡️ v3.x).

It is absolutely fine and appreciable that you and the community around React Query are so prolific, but let's think about the thousands of people and companies that are using RQ in production and need to follow all these changes when a new major release comes out (you may say that no-one is forcing us to introduce the latest version, but... you got the picture).

So, thank you again for everything you give us and... just try not to introduce breaking changes so often 🤓

Cheers!

**Comment by tannerlinsley** [comment]
I will always try to keep things non breaking since I myself maintain large applications that use my libraries too 😉, but at the end of the day, I simply will not hold back a core feature or fix if it warrants breaking changes for too long. While you want stability, many others want features or fixes that will eventually conflict. There’s nothing sacred about major semver increments so long as the changes and migration paths are justified and unless you are a major sponsor of the library or invested/active contributor, it’s not in the library’s best interest to stagnate progress for your sake. 

I know you have the best of intentions with this comment and I even appreciate your positive delivery of it with some compliments 🤩, but as per usual, I will state for posterity that OSS is no place to communicate with entitlement and this comment carries a small amount of that undertone.
